src/pfs/server/server_test.go

```diff
@@ -0,0 +1,165 @@
+	var buffer bytes.Buffer
+	require.NoError(t, pfsutil.GetFile(pfsClient, repo, commit1.Id, "foo", 0, 0, "", nil, &buffer))
+	require.Equal(t, "foo\n", buffer.String())
+	commit2, err := pfsutil.StartCommit(pfsClient, repo, commit1.Id, "")
+	require.NoError(t, err)
+	_, err = pfsutil.PutFile(pfsClient, repo, commit2.Id, "foo", 0, strings.NewReader("foo\n"))
+	require.NoError(t, err)
+	err = pfsutil.FinishCommit(pfsClient, repo, commit2.Id)
+	require.NoError(t, err)
+	buffer = bytes.Buffer{}
+	require.NoError(t, pfsutil.GetFile(pfsClient, repo, commit1.Id, "foo", 0, 0, "", nil, &buffer))
+	require.Equal(t, "foo\n", buffer.String())
+	buffer = bytes.Buffer{}
+	require.NoError(t, pfsutil.GetFile(pfsClient, repo, commit2.Id, "foo", 0, 0, "", nil, &buffer))
+	require.Equal(t, "foo\nfoo\n", buffer.String())
+}
+
+func TestBranch(t *testing.T) {
+	t.Parallel()
+	pfsClient := getPfsClient(t)
+	repo := uniqueString("TestBranch")
+	require.NoError(t, pfsutil.CreateRepo(pfsClient, repo))
+	commit1, err := pfsutil.StartCommit(pfsClient, repo, "", "master")
+	require.NoError(t, err)
+	_, err = pfsutil.PutFile(pfsClient, repo, "master", "foo", 0, strings.NewReader("foo\n"))
+	require.NoError(t, err)
+	require.NoError(t, pfsutil.FinishCommit(pfsClient, repo, "master"))
+	var buffer bytes.Buffer
+	require.NoError(t, pfsutil.GetFile(pfsClient, repo, "master", "foo", 0, 0, "", nil, &buffer))
+	require.Equal(t, "foo\n", buffer.String())
+	branches, err := pfsutil.ListBranch(pfsClient, repo)
+	require.NoError(t, err)
+	require.Equal(t, commit1, branches[0].Commit)
+	require.Equal(t, "master", branches[0].Branch)
+	commit2, err := pfsutil.StartCommit(pfsClient, repo, "", "master")
+	require.NoError(t, err)
+	_, err = pfsutil.PutFile(pfsClient, repo, "master", "foo", 0, strings.NewReader("foo\n"))
+	require.NoError(t, err)
+	err = pfsutil.FinishCommit(pfsClient, repo, "master")
+	require.NoError(t, err)
+	buffer = bytes.Buffer{}
+	require.NoError(t, pfsutil.GetFile(pfsClient, repo, commit1.Id, "foo", 0, 0, "", nil, &buffer))
+	require.Equal(t, "foo\n", buffer.String())
+	buffer = bytes.Buffer{}
+	require.NoError(t, pfsutil.GetFile(pfsClient, repo, "master", "foo", 0, 0, "", nil, &buffer))
+	require.Equal(t, "foo\nfoo\n", buffer.String())
+	branches, err = pfsutil.ListBranch(pfsClient, repo)
+	require.NoError(t, err)
+	require.Equal(t, commit2, branches[0].Commit)
+	require.Equal(t, "master", branches[0].Branch)
+}
+
+func TestDisallowReadsDuringCommit(t *testing.T) {
+	t.Parallel()
+	pfsClient := getPfsClient(t)
+	repo := uniqueString("TestDisallowReadsDuringCommit")
+	require.NoError(t, pfsutil.CreateRepo(pfsClient, repo))
+	commit1, err := pfsutil.StartCommit(pfsClient, repo, "", "")
+	require.NoError(t, err)
+	_, err = pfsutil.PutFile(pfsClient, repo, commit1.Id, "foo", 0, strings.NewReader("foo\n"))
+	require.NoError(t, err)
+
+	// Make sure we can't get the file before the commit is finished
+	var readBuffer bytes.Buffer
```

Comment:
This is a bit OCD but could we have this be the same buffer as below and just reinitialize?